src/rpc/server.ts

```diff
@@ -897,4 +901,110 @@ export class Server {
   public async getVersionInfo(): Promise<Api.GetVersionInfoResponse> {
     return jsonrpc.postObject(this.serverURL.toString(), 'getVersionInfo');
   }
+
+  /**
+   * Returns a contract's balance of a particular SAC asset, if any.
+   *
+   * This is a convenience wrapper around {@link Server.getLedgerEntries}.
+   *
+   * @param {string}  contractId    the contract ID (string `C...`) whose
+   *    balance of `sac` you want to know
+   * @param {Asset}   sac     the built-in SAC token (e.g. `USDC:GABC...`) that
+   *    you are querying from the given `contract`.
+   * @param {string}  [networkPassphrase] optionally, the network passphrase to
+   *    which this token applies. If omitted, a request about network
+   *    information will be made (see {@link getNetwork}), since contract IDs
+   *    for assets are specific to a network. You can refer to {@link Networks}
+   *    for a list of built-in passphrases, e.g., `Networks.TESTNET`.
+   *
+   * @returns {Promise<Api.BalanceResponse>}, which will contain the balance
+   *    entry details if and only if the request returned a valid balance ledger
+   *    entry. If it doesn't, the `balanceEntry` field will not exist.
+   *
+   * @throws {TypeError} If `contractId` is not a valid contract strkey (C...).
+   *
+   * @see getLedgerEntries
+   * @see https://developers.stellar.org/docs/tokens/stellar-asset-contract
+   *
+   * @example
+   * // assume `contractId` is some contract with an XLM balance
+   * // assume server is an instantiated `Server` instance.
+   * const entry = (await server.getSACBalance(
+   *   new Address(contractId),
+   *   Asset.native(),
+   *   Networks.PUBLIC
+   * ));
+   *
+   * // assumes BigInt support:
+   * console.log(
+   *   entry.balanceEntry ?
+   *   BigInt(entry.balanceEntry.amount) :
+   *   "Contract has no XLM");
+   */
+  public async getSACBalance(
```

Comment:
It's interesting to see this function on the rpc.Server class, given that the RPC Server doesn't provide a `getSACBalance` function itself.

Early on in the RPC-CLI relationship we built a ton of helper functionality similar to this into the Rust RPC client (equivalent to the rpc.Server here in the JS SDK). We regretted it, the boundaries between what a client needed to do vs application logic got intertwined fast. We have slowly been stripping the Rust RPC client back to be a thing wrapper around the actual RPC API because of that. This situation isn't quite the same, but it's similar.

What's the vision for where the responsibilities of the rpc.Server begin and end?

If we plan to add more utility functionality like this, that is supporting one specific contract, will it continue to go onto the rpc.Server class?

Maybe this function belongs in a class or on a type/module focused on the SAC, where you give it an RPC Client (called the rpc.Server here), and then it does those additional functionality.

Reply:
You make an interesting point @leighmcculloch, but we already have tons of utility functions like this. If you look at [the methods](https://github.com/stellar/js-stellar-sdk/blob/92dbfb7a15d6a7960be2e604b9f587ee1e026144/src/rpc/server.ts), you'll see `getContractData`, `requestAirdrop`, heck, even `getAccount`. All of these have no 1:1 relation to actual Stellar RPC endpoints. This is largely a product of the fact that very common interactions are very unintuitive (due to XDR complexities, special data entry formats, etc.) and expecting users to write their own is just... mean :laughing: There's also https://github.com/stellar/js-stellar-sdk/issues/1003 which hasn't been defined at all but it's interpretation (to me) is fairly clear: people shouldn't need to know how to hand-craft the perfect `LedgerKey` for common functionality.

I think it's totally okay to provide these convenience layers at the `Server` layer because they're universal (the "app code bungling client code" issue hasn't cropped up here). Maybe they could belong in their own `EnhancedServer` variant, but who wants to adapt their code to that?

> What's the vision for where the responsibilities of the rpc.Server begin and end?

In my mind, it's "whatever it takes for people to interact with the RPC easily." I understand that that's extremely broad and easy to scope creep in, but I/we only really add stuff that has clearly defined, common use cases, shouldn't be repeated across `N` codebases in the ecosystem, and hides esoteric knowledge about Soroban internals from the user.

Reply:
The boundary can be subtle, nothing wrong with that. I expect the same problems won't show up here because no app code is bundled close to the JS SDK. It's probably more a lot why things went awry in the Rust SDK. 

👍

Reply:
Yep, good point on lack of bundling (for now... :shushing_face:)!